src/core/components/bootstrap.js

```diff
@@ -1,9 +1,24 @@
 'use strict'
 
 const defaultNodes = require('../runtime/config-nodejs.json').Bootstrap
-const MultiAddr = require('multiaddr')
+const Multiaddr = require('multiaddr')
 const promisify = require('promisify-es6')
 
+function isValid (ma) {
+  if (typeof ma === 'string') {
+    try {
+      ma = new Multiaddr(ma)
+      return Boolean(ma)
```

Comment:
This idea comes from type checking discussion. It would be cool if `isValid()` functions would return `false` if invalid and return a valid object whatever got checked (in this case a `Multiaddr`).

That way you could also use `isValid()` to make sure you always get a valid `Multiaddr` object, no matter if it was already one or of if it was a string.

Reply:
This got upgraded in https://github.com/ipfs/js-ipfs/pull/1227